config/labels.go

```diff
@@ -64,12 +63,20 @@ const LABEL_IS_BACKUP = "is-backup"
 const LABEL_STARTUP = "startup"
 const LABEL_SHUTDOWN = "shutdown"
 
-const LABEL_MINOR_UPGRADE = "minor-upgrade"
-const LABEL_UPGRADE_IN_PROGRESS = "upgrade-in-progress"
-const LABEL_UPGRADE_COMPLETED = "upgrade-complete"
-const LABEL_UPGRADE_REPLICA = "upgrade-replicas"
-const LABEL_UPGRADE_PRIMARY = "upgrade-primary"
-const LABEL_UPGRADE_BACKREST = "upgrade-backrest"
+//label for the pgcluster upgrade
+const LABEL_UPGRADE = "upgrade"
+
+//label indicating the pgcluster requires an upgrade
+const LABEL_IS_UPGRADED = "is-upgraded"
+
+//label showing upgrade from and to information
+const LABEL_UPGRADE_INFO = "upgrade-info"
+
+// label for the pgbadger port that can be lost during an upgrade
+const LABEL_PGBADGER_PORT = "pgbadger-port"
```

Comment:
Isn't this stored in the `pgcluster` spec under `pgbadgerport`?  Curious why it would/could be lost.

Reply:
it should normally be, yes. If you look at line 260 in operator/cluster/upgrade.go, you'll see those values are preferred. These labels are used by `CreateUpgrade` in apiserver/upgradeservice/upgradeimpl.go pgtask to pass the default port values from the apiserver side via the pgtask (to be used later, similar to how they are used here: 
https://github.com/CrunchyData/postgres-operator/blob/c9759b8d68f4758dcea3fb8f823e46da76567838/apiserver/clusterservice/clusterimpl.go#L1061-L1062 )
If there is a better way to grab the default config values from the apiserver side, I'm happy to update them. This seemed to be the least intrusive option at the time.

Reply:
Looking at `CreateUpgrade`, I see that the badger port is set to the value defined in the `pgo.yaml`, e.g.:

```go
spec.Parameters[config.LABEL_PGBADGER_PORT] = apiserver.Pgo.Cluster.PGBadgerPort
```

And then in `upgrade.go` the port in the `pgcluster` always takes precedence over what is in the parameters, e.g. the parameter is only used when there is no port in the spec:

```go
if pgcluster.Spec.PGBadgerPort == "" {
    pgcluster.Spec.PGBadgerPort = parameters[config.LABEL_PGBADGER_PORT]
}
```

So that def makes sense.

I'm just wondering if we even need to add the `pgo.yaml` values for these ports to the `pgtask` at all, since we can just access the contents of the `pgo.yaml` when processing the `pgtask`, e.g.:

```go
if pgcluster.Spec.PGBadgerPort == "" {
    pgcluster.Spec.PGBadgerPort = operator.Pgo.Cluster.PGBadgerPort
}
```

It seems as though the ports would only need to be explicitly specified in the `pgtask` if there was a need to override the contents of the `pgcluster` and `pgo.yaml` (which I'm thinking is def out of scope here).

Also, just curious...is there actually ever an instance when these ports would _not_ be found on the `pgcluster`?

Reply:
Thanks for the suggestions!

I definitely agree that `operator.Pgo.Cluster.PGBadgerPort` makes more sense. I'm really struggling to remember what error/reason I ran into for passing the value over from the apiserver side. Perhaps it was related to some instantiation errors I hit when first starting? 
In any case, I'll update the code to this method and test to make sure it still works as expected.

I do seem to remember a problem where the I had an error related to the port on the `pgcluster` CRD, but that may have been related to an ingestion problem I've since solved. Either way, I think the check for the empty value is worth keeping, even if the parameter storage message currently used is not.

Reply:
the update to operator.Pgo.Cluster.PGBadgerPort worked as you described in testing. I've updated the code to use that and removed the parameter passing from CreateUpgrade in upgradeimpl.go. I have left the checks in place just in case that nil issue should come up again as that check is fairly lightweight at this point. Thanks again!